### ISSUE TITLE
[bug]: Advanced Recovery Sync Issue - Voicemail light is blinking on the phone when there is no message on Primary Server

### ISSUE THREAD

**Issue by jasonurkow** [FreePBX/issue-tracker#458]
### FreePBX Version

FreePBX 17

### Issue Description

Hello,

The Message Waiting Indicator (MWI) light is blinking, indicating there is a voicemail, even though there are no messages. When they dial into the voicemail system, it says there are no new messages, and the light turns off after they hang up. However, the light reappears randomly within 30 minutes to 24 hours.

I also have Advanced Recovery, and I'm wondering if the synced messages on the backup server could be triggering the MWI light on the phone?

### Operating Environment

PBX Version: 17.0.19.12
PBX Distro: 12.7.8-2402-3.sng12
Asterisk Version: 21.4.1
Advanced Recovery: 17.0.4

**Comment by jasonurkow** [comment]
So after a bunch of testing, this is what I have found.

I have Advanced Recovery installed, I have my Primary Server and my Secondary Server.

**Scenario 1 (Working):**

- A voicemail is received.
- The phone receives a WMI message from the Primary Server, indicating a new message, and the Message Waiting Indicator (MWI) light illuminates.
- After an hour, the voicemail is accessed and deleted.
- The phone receives a "no MWI" message from the Primary Server, and the MWI light extinguishes.

**Scenario 2 (MWI Light Remains Illuminated):**

- A voicemail is received.
- The phone receives a WMI message from the Primary Server, and the MWI light illuminates.
- **_After a day (following overnight voicemail replication to the Secondary Server),_** the voicemail is accessed and deleted.
- The phone receives a "no MWI" message from the Primary Server, and the MWI light extinguishes.
- However, approximately 25 minutes later, the phone receives an MWI message from the Secondary Server, and the MWI light re-illuminates.
- Upon checking for voicemails, none are found.
- Verification on the Primary Server shows no voicemails for the extension, but the Secondary Server still retains the deleted voicemail.

**Investigation:**

I attempted a manual sync within Advanced Recovery, but the voicemail remained on the Secondary Server.  If I clear the inbox on the secondary server from the admin console, the light extinguishes.

**Potential Issue:**

It appears that the Secondary Server in Advanced Recovery is not properly deleting voicemails after they are accessed and deleted on the Primary Server and replicated.

**Request:**

Is syncing of voicemails a known issue within Advanced Recovery?

Thank you.

Current PBX Version:    17.0.19.23
Advanced Recovery:     17.0.4
Voicemail:                     17.0.5.29
Phone Model:               Grandstream GRP2615

**Comment by blazestudios97** [comment]
This isn't a bug, this is the result of how your particular setup is being done. I'm going to guess that you are using the primary server/secondary server options in the phone which registers the phone to both servers at once, including MWI.

**Comment by dolesec** [comment]
I've encountered this before using warm spare and Polycom endpoints. I'll have to look back at the particular fix. If I recall correctly, it involved changing the extension's MWI method and editing the poly config's base file. 

Depending on your requirements,  voicemail to email with delete could also solve the issue.

**Comment by blazestudios97** [comment]
> I've encountered this before using warm spare and Polycom endpoints. I'll have to look back at the particular fix. If I recall correctly, it involved changing the extension's MWI method and editing the poly config's base file.
> 
> Depending on your requirements, voicemail to email with delete could also solve the issue.

Right but did you do the same thing? Did you use the primary/secondary server options? Because when you do that, you are registering the phone to both servers at once. I use FQDNs and SRV records for the FQDNs. The phones are then registered to the primary server **only** and when that primary server goes down or becomes unresponsive, the phone starts using the backup. In this case, I never see MWI from the secondary server when on the primary.

It doesn't matter how often you do a sync, if you delete voicemail from the primary until Advanced Recovery does a sync with the secondary any and all voicemail since the previous sync will still exist on the secondary. If you are registering to both servers at once the only way to keep the MWI for voicemail in sync would be to sync the systems each and every time a voicemail is left or modified.

Advanced Recovery isn't a cluster solution is a failover/redundancy solution. It definitely isn't a real-time sync solution.

**Comment by jasonurkow** [comment]
You are correct I do have both primary and secondary servers registered on
my phones. But the problem is that the secondary server even with the sync
will never delete any voicemails and I have to go in every once in awhile
and delete all voicemails off the secondary server otherwise it fills up.

**Comment by blazestudios97** [comment]
Do you run an automatic "Apply Config" when the sync is completed?

Or are you saying that extension 100 has 0 voicemail on primary, you do the sync on secondary and the secondary still has all the old voicemails present that existed before the sync?

**Comment by jasonurkow** [comment]
That's correct on your second scenario. If an extension has zero voicemails
currently on the primary server but The secondary server has three
voicemails on it due to it being synced up in the past with voicemails,
when the sync happens those voicemails on the secondary server are not
removed. And I don't recall if I have a auto apply happen after a sync, how
can I check if that happens or not?  Just because I don't recall the apply
button being up every time I go on the secondary server after a sync, so it
must be applying?

**Comment by mgajjar19** [comment]
I use Unison with crontab as a workaround to fix the voicemail light issue.
Unison syncs the voicemail directory between both servers, and I have it set at every 5 minutes.

I am still testing it thoroughly, but you can give it a try.

**Comment by jissphilip** [comment]
HI , 

Please try with below versions 
voicemail 16.0.63
voicemail 17.0.5.32
